tasks/Task.cpp

```diff
@@ -0,0 +1,238 @@
+{
+    // Creating the driver object
+    driver.reset(new Driver());
+    if (!_io_port.get().empty())
+        driver->openURI(_io_port.get());
+    setDriver(driver.get());
+
+    if (! TaskBase::configureHook())
+        return false;
+
+    DeviceSettings settings = _device_settings.get();
+    DeviceSettings current_settings = driver->getCurrentSetting();
+    if(_change_parameters.get())
+    {
+        //TODO need validation
+        updateDeviceParameters(settings, current_settings);
+    }
+
+    VersionNumbers device_info = driver->getFirmwareInformation();
+    if(device_info.firmwareVersion.find("1.7") == std::string::npos)
+        RTT::log(RTT::Error) << "Component was developed for firmware version \"1.7\" and actual version is: \""<< device_info.firmwareVersion <<"\". Be aware of eventual incompatibility." << std::endl;
+    RTT::log(RTT::Info) << "USBL's firmware information: Firmware version: \""<< device_info.firmwareVersion <<"\"; Physical and Data-Link layer protocol: \""<<
+            device_info.accousticVersion <<"\"; Manufacturer: \""<< device_info.manufacturer << "\"" << std::endl;
+
+    return true;
+}
+bool Task::startHook()
+{
+    if (! TaskBase::startHook())
+        return false;
+    return true;
+}
+void Task::updateHook()
+{
+    TaskBase::updateHook();
+}
+void Task::errorHook()
+{
+    TaskBase::errorHook();
+}
+void Task::stopHook()
+{
+    TaskBase::stopHook();
+}
+void Task::cleanupHook()
+{
+    TaskBase::cleanupHook();
+}
+void Task::processIO()
+{
+
+    _device_settings.set(driver->getCurrentSetting());
+
+    _acoustic_channel.write(getAcousticChannelparameters());
+
+    AcousticConnection acoustic_connection = driver->getConnectionStatus();
+    _acoustic_connection.write(acoustic_connection);
+
+    // TODO define exactly what to do for each acoustic_connection status
+    if(acoustic_connection.status == ONLINE || acoustic_connection.status == OFFLINE_READY || acoustic_connection.status == INITIATION_LISTEN )
+    {
+        if(_message_input.read(send_IM) == RTT::NewData)
+            driver->sendInstantMessage(send_IM);
+
+        iodrivers_base::RawPacket raw_data_input;
+        if(_raw_data_input.read(raw_data_input) == RTT::NewData)
+        {
+            if(driver->getMode() == DATA)
+                driver->sendRawData(raw_data_input.data);
+            else
+                RTT::log(RTT::Error) << "Device can not send raw_data \""<< raw_data_input.data.data() << "\" in COMMAND mode. Be sure to switch device to DATA mode" << std::endl;
+        }
+    }
+
+    while(driver->hasNotification())
+        processNotification(driver->getNotification());
+
+    while(driver->hasRawData())
+    {
+        iodrivers_base::RawPacket raw_packet_buffer;
+        raw_packet_buffer.time = base::Time::now();
+        raw_packet_buffer.data = driver->getRawData();
+        _raw_data_output.write(raw_packet_buffer);
+    }
+
+    if(acoustic_connection.status == OFFLINE_ALARM)
```

Comment:
It appear when an internal error has occurred on device. Manual says to reset the device.